homeassistant/components/binary_sensor/maxcul.py

```diff
@@ -0,0 +1,107 @@
+"""
+Support for MAX! shutter contacts.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.maxcul/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import dispatcher_connect
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, DEVICE_CLASSES, PLATFORM_SCHEMA
+)
+from homeassistant.const import (
+    CONF_ID, CONF_DEVICES, CONF_DEVICE_CLASS
+)
+from homeassistant.components.maxcul import (
+    DATA_MAXCUL, SIGNAL_SHUTTER_UPDATE
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['maxcul']
+
+DEFAULT_DEVICE_CLASS = 'opening'
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ID): cv.positive_int,
+    vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS):
+    vol.In(DEVICE_CLASSES),
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICES): vol.Schema({
+        cv.string: DEVICE_SCHEMA
+    })
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the maxcul binary sensor platform."""
+    devices = [
+        MaxShutterContact(
+            hass,
+            device[CONF_ID],
+            name,
+            device[CONF_DEVICE_CLASS]
+        )
+        for name, device
+        in config[CONF_DEVICES].items()
+    ]
+    add_devices(devices)
+
+
+class MaxShutterContact(BinarySensorDevice):
+    """Representation of a MAX! shutter contact."""
+
+    def __init__(self, hass, device_id, name,
```

Comment:
I'd prefer if you only pass the maxcul_handle instead of `hass`. `hass` will be set on the entity when it has been added to home assistant.